src/System.Device.Gpio/System/Device/I2c/I2cBus.cs

```diff
@@ -0,0 +1,83 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Device.I2c
+{
+    /// <summary>
+    /// I2C bus communication channel.
+    /// </summary>
+    public abstract partial class I2cBus : IDisposable
+    {
+        /// <summary>
+        /// Creates default I2cBus
+        /// </summary>
+        /// <param name="busId">The bus ID.</param>
+        /// <returns>I2cBus instance.</returns>
+        public static I2cBus Create(int busId)
+        {
+            if (Environment.OSVersion.Platform == PlatformID.Win32NT)
+            {
+                return CreateWindows10I2cBus(busId);
+            }
+            else
+            {
+                return UnixI2cBus.Create(busId);
+            }
+        }
+
+        /// <summary>
+        /// Creates I2C device.
+        /// </summary>
+        /// <param name="deviceAddress">Device address related with the device to create.</param>
+        /// <returns>I2cDevice instance.</returns>
+        public abstract I2cDevice CreateDevice(int deviceAddress);
+
+        /// <summary>
+        /// Removes I2C device.
+        /// </summary>
+        /// <param name="deviceAddress">Device address to create</param>
+        public abstract void RemoveDevice(int deviceAddress);
```

Comment:
This should probably have an overload taking an I2cDevice.

Reply:
I think we should do one or the other, not both. In the issue I've proposed both solutions and address made more sense as it was similar to GPIO APIs

Reply:
I agree with @krwq and prefer the device address as it's what's used and how the I2C is managed all up. This makes for me the moist sense.